world/templates/models.py

```diff
@@ -0,0 +1,82 @@
+"""
+App that handles the creation, management and assignment of Templates, which are wrappers around
+any arbitrary text into a format that can be injected into any 'desc' in order to do the following:
+
+1) Reduce database bloat by making the text 'loadable' via markup.
+2) Allowing users to manage their own templates and, because it uses markup, study won't let them steal templates.
+3) Allows for sharing of templates and attribution (or not as desired)
+
+"""
+
+from django.db import models
+from evennia.typeclasses.models import SharedMemoryModel
+from .template_manager import TemplateManager
+
+
+class Template(SharedMemoryModel):
+    """
+    A model for storing raw text data (ASCII) in a manageable format. Each
+    Template will be owned by an individual player account.
+
+    There are certain design implications as it regards templates:
+
+        1) They are immutable upon creation. (This will prevent weird issues
+           when markup is parsed in validly created templates.)
+        2) Use of templates MUST be parsed on ingest, for the same reasons as
+           above.
+        3) Templates will, by default be private.
+    """
+
+    PRIVATE = 'PR'
+    RESTRICTED = 'RS'
+    OPEN = 'OP'
+    ACCESS_LEVELS = (
+        (PRIVATE, 'PRIVATE'),
+        (RESTRICTED, 'RESTRICTED'),
+        (OPEN, 'OPEN')
+    )
+
+    owner = models.ForeignKey('character.PlayerAccount', related_name='templates', db_index=True)
+    desc = models.TextField()
+
+    access_level = models.CharField(max_length=2, choices=ACCESS_LEVELS, default=PRIVATE)
+
+    attribution = models.CharField(max_length=60)
+    apply_attribution = models.BooleanField(default=False)
+
+    title = models.CharField(max_length=255)
+
+    applied_to = models.ManyToManyField('objects.ObjectDB', blank=True)
+
+    grantees = models.ManyToManyField('character.RosterEntry', through="TemplateGrantee", blank=True)
+
+    accessible = TemplateManager()
```

Comment:
Not a big deal, but I prefer managers that add methods rather than changing the default queryset replace 'objects' so that they're used by default and I don't need to remember the manager name as well as their methods.
```suggestion
    objects = TemplateManager()
```

Reply:
Resolved in new push.